CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+- Maintainers have the necessary context in order to support and review contributions.
+
+## PR Naming
+
+PRs should be titled as following: 
+```txt
+pkg: Concise title of PR
+```
+
+**Example:**
+
+```txt
+service/header: Remove race in core_listener
+```
+
+## Changelog
+
+Every *notable* fix, improvement, feature, or breaking change should be made in a
+pull-request that includes an update to the `CHANGELOG_PENDING.md` file.
+
+Changelog entries should be formatted as follows:
+
+```md
+- [module/pkg: Some description about the change #xxx](link to PR) [@contributor](link to contributer github) 
+```
+
+Here, `module` is the part of the code that changed (typically a
+top-level Go package), `xxx` is the pull-request number, and `contributor`
+is the author/s of the change.
+
+## Branching Model and Release
+
+The main development branch is main.
+
+Every release is maintained in a release branch named `vX.Y.Z`.
+
+Note all pull requests should be squash merged except for merging to a release branch (named `vX.Y`). This keeps the commit history clean and makes it
+easy to reference the pull request where a change was introduced.
+
+### Development Procedure
+
+The latest state of development is on `main`, which must never fail `make test`. _Never_ force push `main`.
+
+To begin contributing, create a development branch on your fork.
+
+Make changes, and before submitting a pull request, update the `CHANGELOG_PENDING.md` to record your change. Also, run either `git rebase` or `git merge` on top of the latest `main`. (Since pull requests are squash-merged, either is fine!)
+
+Sometimes (often!) pull requests get out-of-date with main, as other people merge different pull requests to main. It is our convention that pull request authors are responsible for updating their branches with `main`. (This also means that you shouldn't update someone else's branch for them; even if it seems like you're doing them a favor, you may be interfering with their git flow in some way!)
+
+#### Merging Pull Requests
+
+It is also our convention that authors merge their own pull requests, when possible. External contributors may not have the necessary permissions to do this, in which case, a member of the core team will merge the pull request once it's been approved.
+
+Before merging a pull request:
+
+- Ensure pull branch is up-to-date with a recent `main` (GitHub won't let you merge without this!)
+- Run `make test` to ensure that all tests pass
+- [Squash](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) merge pull request
+
+### Git Commit Style
+
+We follow the [Go style guide on commit messages](https://tip.golang.org/doc/contribute.html#commit_messages). Write concise commits that start with the package name and have a description that finishes the sentence "This change modifies Celestia-Node to...". For example,
+
+```sh
+cmd/debug: execute p.Signal only when p is not nil
```

Comment:
We should also encourage people to mark the type of the change, e.g. `feat/perf/fix`, etc, like `feat(service/header): new something to do something.` to also follow https://www.conventionalcommits.org/en/v1.0.0/